spyder/app/tests/test_mainwindow.py

```diff
@@ -205,6 +217,40 @@ def test_calltip(main_window, qtbot):
     main_window.editor.close_file()
 
 
+@pytest.mark.slow
+@flaky(max_runs=3)
+def test_change_directory_in_project_explorer(main_window, qtbot, tmpdir):
```

Comment:
This test doesn't need to be interact with other plugins (e.g. the IPython console), so it doesn't belong here. Please move it to `spyder/plugins/tests/test_projects.py`.